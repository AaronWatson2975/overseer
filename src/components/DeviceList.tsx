import {
  Button,
  CircularProgress,
  List,
  ListSubheader,
} from "@material-ui/core";
import React, { Component } from "react";
import { IAndroidDevice } from "../interfaces";
import { fetchDevices } from "../models";
import { Device } from "./Device";

interface IProps {
  devices: IAndroidDevice[];
  hasDevices: boolean;
}

class DeviceList extends Component<IProps> {
  public render() {
    return (
      <>
        <List
          component="nav"
          aria-labelledby="nested-list-subheader"
          subheader={
            <ListSubheader component="div" id="nested-list-subheader">
              Connected Android Devices
            </ListSubheader>
          }
        >
          {!this.props.hasDevices && (
            <CircularProgress className="centered-progress" />
          )}
          {this.props.devices.map(device => (
            <Device device={device} />
          ))}
        </List>
        <Button variant="outlined" color="primary">
          Connect to IP
        </Button>
      </>
    );
  }
}

export { DeviceList };
