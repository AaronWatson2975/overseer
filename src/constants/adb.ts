export const DEFAULT_PORT = 5555;
