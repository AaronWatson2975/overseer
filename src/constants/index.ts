export * from "./mock.devices";
export * from "./adb";
