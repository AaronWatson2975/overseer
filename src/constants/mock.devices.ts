import { AndroidDevice, Connection } from "../enums";
import { IAndroidDevice } from "../interfaces";

/**
 * Data interface
 *
 * make - The make of the Androiddevice
 * model - The model of the Androiddevice
 * serial - The serial number of the Androiddevice
 * connection - The way the Androiddevice is connected
 * type - The type of Androiddevice(phone, tablet, etc.)
 * ip - The ip address of the Androiddevice
 * duplicate - Whether another Androiddevice of the same
 * make and model exists in the collection
 */

const mockDevices: IAndroidDevice[] = [
  {
    make: "Dragonboard",
    model: "820c",
    serial: "c0ffee11",
    connection: Connection.USB,
    type: AndroidDevice.DevBoard,
    ip: "127.0.0.1",
    duplicate: false,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel 3a",
    serial: "a54bde3d",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.45",
    duplicate: true,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel 3a",
    serial: "76cda2ea",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.48",
    duplicate: true,
    offline: false
  },
  {
    make: "Google",
    model: "Pixel C",
    serial: "1234abcd",
    connection: Connection.USB,
    type: AndroidDevice.Tablet,
    ip: "127.0.0.1",
    duplicate: false,
    offline: false
  },
  {
    make: "Inforce",
    model: "6640",
    serial: "c0ffee22",
    connection: Connection.Network,
    type: AndroidDevice.DevBoard,
    ip: "192.168.2.15",
    duplicate: false,
    offline: false
  },
  {
    make: "Samsung",
    model: "Galaxy S6",
    serial: "68cd4ef2",
    connection: Connection.USB,
    type: AndroidDevice.Phone,
    ip: "192.168.2.13",
    duplicate: false,
    offline: false
  },
  {
    make: "Samsung",
    model: "Galaxy S6",
    serial: "68cd4ef2",
    connection: Connection.Network,
    type: AndroidDevice.Phone,
    ip: "192.168.2.13",
    duplicate: false,
    offline: false
  },
  {
    make: "Raspberry Pi",
    model: "3b",
    serial: "c0ffee32",
    connection: Connection.Network,
    type: AndroidDevice.DevBoard,
    ip: "192.168.2.76",
    duplicate: false,
    offline: true
  }
];

export { mockDevices };
