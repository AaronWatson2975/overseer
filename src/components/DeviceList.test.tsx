import { shallow } from "enzyme";
import React from "react";
import { DeviceList } from "./DeviceList";
import { mockDevices } from "../constants";

describe("doesn't crash", () => {
  it("renders without crashing", () => {
    shallow(<DeviceList devices={mockDevices} hasDevices={true} />);
  });
});

test("basic snapshot test", () => {
  const wrapper = shallow(
    <DeviceList devices={mockDevices} hasDevices={true} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("zero state", () => {
  const wrapper = shallow(<DeviceList devices={[]} hasDevices={true} />);
  expect(wrapper).toMatchSnapshot();
});

test("loading", () => {
  const wrapper = shallow(<DeviceList devices={[]} hasDevices={false} />);
  expect(wrapper).toMatchSnapshot();
});
