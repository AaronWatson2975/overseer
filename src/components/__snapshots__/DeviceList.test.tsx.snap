// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot test 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List))
    aria-labelledby="nested-list-subheader"
    component="nav"
    subheader={
      <ForwardRef(WithStyles)
        component="div"
        id="nested-list-subheader"
      >
        Connected Android Devices
      </ForwardRef(WithStyles)>
    }
  >
    <Device
      device={
        Object {
          "connection": 1,
          "duplicate": false,
          "ip": "127.0.0.1",
          "make": "Dragonboard",
          "model": "820c",
          "offline": false,
          "serial": "c0ffee11",
          "type": 0,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 1,
          "duplicate": true,
          "ip": "192.168.2.45",
          "make": "Google",
          "model": "Pixel 3a",
          "offline": false,
          "serial": "a54bde3d",
          "type": 1,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 1,
          "duplicate": true,
          "ip": "192.168.2.48",
          "make": "Google",
          "model": "Pixel 3a",
          "offline": false,
          "serial": "76cda2ea",
          "type": 1,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 1,
          "duplicate": false,
          "ip": "127.0.0.1",
          "make": "Google",
          "model": "Pixel C",
          "offline": false,
          "serial": "1234abcd",
          "type": 2,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 0,
          "duplicate": false,
          "ip": "192.168.2.15",
          "make": "Inforce",
          "model": "6640",
          "offline": false,
          "serial": "c0ffee22",
          "type": 0,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 1,
          "duplicate": false,
          "ip": "192.168.2.13",
          "make": "Samsung",
          "model": "Galaxy S6",
          "offline": false,
          "serial": "68cd4ef2",
          "type": 1,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 0,
          "duplicate": false,
          "ip": "192.168.2.13",
          "make": "Samsung",
          "model": "Galaxy S6",
          "offline": false,
          "serial": "68cd4ef2",
          "type": 1,
        }
      }
    />
    <Device
      device={
        Object {
          "connection": 0,
          "duplicate": false,
          "ip": "192.168.2.76",
          "make": "Raspberry Pi",
          "model": "3b",
          "offline": true,
          "serial": "c0ffee32",
          "type": 0,
        }
      }
    />
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    variant="outlined"
  >
    Connect to IP
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;

exports[`loading 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List))
    aria-labelledby="nested-list-subheader"
    component="nav"
    subheader={
      <ForwardRef(WithStyles)
        component="div"
        id="nested-list-subheader"
      >
        Connected Android Devices
      </ForwardRef(WithStyles)>
    }
  >
    <WithStyles(ForwardRef(CircularProgress))
      className="centered-progress"
    />
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    variant="outlined"
  >
    Connect to IP
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;

exports[`zero state 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List))
    aria-labelledby="nested-list-subheader"
    component="nav"
    subheader={
      <ForwardRef(WithStyles)
        component="div"
        id="nested-list-subheader"
      >
        Connected Android Devices
      </ForwardRef(WithStyles)>
    }
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    variant="outlined"
  >
    Connect to IP
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;
